### PR TITLE
Introduction of AttributeDouble from LFRic-Lite

### PR DESCRIPTION
**Pull request by phlndrwd**
This PR introduces the equivalent fix as in LFRic-Lite's recent bugfix: https://github.com/JCSDA-internal/lfric-lite-jedi/pull/316

Also simplifies the use of constants by dropping the namespace prefix where applicable (this is why a larger number of files are impacted by this PR than strictly necessary).

Current testing outputs: http://fcm1/cylc-review/taskjobs/punderwo/?suite=monio_dbl_attr_02

### CODE REVIEW

File: src/monio/AttributeDouble.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+#pragma once
+
+#include <string>
+
+#include "AttributeBase.h"
+
+namespace monio {
+/// \brief Concrete class for integer attributes of a NetCDF file
```
Comment (mo-joshuacolclough) [inline src/monio/AttributeDouble.h:14]:
```suggestion
/// \brief Concrete class for double attributes of a NetCDF file
```

File: src/monio/AttributeDouble.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class AttributeDouble : public AttributeBase {
+ public:
+  AttributeDouble(const std::string& name, const double value);
+
+  AttributeDouble()                                  = delete;  //!< Deleted default constructor
+  AttributeDouble(AttributeDouble&&)                 = delete;  //!< Deleted move constructor
+  AttributeDouble(const AttributeDouble&)            = delete;  //!< Deleted copy constructor
+  AttributeDouble& operator=(AttributeDouble&&)      = delete;  //!< Deleted move assignment
+  AttributeDouble& operator=(const AttributeDouble&) = delete;  //!< Deleted copy assignment
```
Comment (mo-joshuacolclough) [inline src/monio/AttributeDouble.h:23]: Looks good, more of a question on my part: if the base class is not copyable/movable, is this needed? I guess is it for readability?

File: src/monio/File.cc [hunk 3]
```diff
@@ -128,16 +129,24 @@ void monio::File::readVariable(Metadata& metadata, netCDF::NcVar ncVar) {
...
-  else if (varType == netCDF::NcType::nc_FLOAT)
-    var = std::make_shared<Variable>(varName, consts::eFloat);
-  else if (varType == netCDF::NcType::nc_DOUBLE)
-    var = std::make_shared<Variable>(varName, consts::eDouble);
-  else
-    throw std::runtime_error("File::readVariable()> Variable data type " +
-                                varType.getName() + " not coded for.");
 
+  switch (varType.getId()) {
+    case netCDF::NcType::nc_DOUBLE: {
```
Comment (mo-joshuacolclough) [inline src/monio/File.cc:134]: Nice 😎

**Review by mo-joshuacolclough — APPROVED** [review]
Thanks for the fix. Not familiar with the codebase (yet!) but the changes look good. Had a few Qs ->

File: src/monio/AttributeDouble.h (hunk 1 above)
Comment (phlndrwd) [reply src/monio/AttributeDouble.h:14]: Good spot. I'll put that change in the next PR

File: src/monio/AttributeDouble.h (hunk 2 above)
Comment (phlndrwd) [reply src/monio/AttributeDouble.h:23]: Yeah, mostly to be completely explicit. I think it's possible to delete certain constructors/assignments in the base and to define them in the derived class. Though it does produce problems. I just put that in to make the intent clear.

**Pull request merged**